src/main/java/no/nav/familie/integrasjoner/client/rest/PdlRestClient.kt

```diff
@@ -22,22 +22,28 @@ class PdlRestClient(@Value("\${PDL_URL}") pdlBaseUrl: URI,
     : AbstractRestClient(restTemplate, "pdl.personinfo") {
 
     private val pdlUri = UriUtil.uri(pdlBaseUrl, PATH_GRAPHQL)
-    private val graphQL = this::class.java.getResource("/pdl/hentperson.graphql").readText().graphqlCompatible()
 
-    fun hentPerson(personIdent: String, tema: String): Person {
+    fun hentPerson(personIdent: String, tema: String, personInfoQuery: PersonInfoQuery): Person {
+
         val pdlPersonRequest = PdlPersonRequest(variables = PdlPersonRequestVariables(personIdent),
-                                                query = graphQL)
+                                                query = personInfoQuery.graphQL)
         try {
             val response = postForEntity<PdlHentPersonResponse>(pdlUri,
                                                                 pdlPersonRequest,
                                                                 httpHeaders(tema))
             if (response != null && !response.harFeil()) {
                 return Result.runCatching {
-                    val familierelasjoner: Set<Familierelasjon> = response.data?.person!!.familierelasjoner.map { relasjon ->
-                        Familierelasjon(personIdent = Personident(id= relasjon.relatertPersonsIdent),
-                                        relasjonsrolle = relasjon.relatertPersonsRolle.toString())
-                    }.toSet()
-                    response.data.person.let {
+                    val familierelasjoner: Set<Familierelasjon> =
+                            when (personInfoQuery) {
+                                PersonInfoQuery.ENKEL -> emptySet()
+                                PersonInfoQuery.MED_RELASJONER -> {
+                                    response.data?.person!!.familierelasjoner.map { relasjon ->
+                                        Familierelasjon(personIdent = Personident(id = relasjon.relatertPersonsIdent),
+                                                        relasjonsrolle = relasjon.relatertPersonsRolle.toString())
+                                    }.toSet()
+                                }
+                            }
+                    response.data?.person!!.let {
```

Comment:
Litt mye bruk av ? og !!. Kan det skrives om til å være litt mer robust med bedre feilmeldinger kanskje?

Reply:
Etter nærmere ettersyn så tror jeg vi kan sette data-feltet som not null. Jeg kan endre på det. Person vet vi egentlig at vi skal ha på dette stedet i koden ettersom den kjører inne i en if-block der vi har sjekket om det er feil på respons-objektet. Er det ikke det så skal person være satt.